### PR TITLE
ASoC: SOF: update sof_ipc_pipe_new

### PR DESCRIPTION
**Pull request by tlauda**
Updates sof_ipc_pipe_new struct to align with changes
on FW side. Only field names are changing.

Signed-off-by: Tomasz Lauda <tomasz.lauda@linux.intel.com>

### CODE REVIEW

File: include/sound/sof/topology.h [hunk 1]
```diff
@@ -218,21 +218,27 @@ struct sof_ipc_comp_reply {
...
+	uint32_t period;	/**< execution period in us*/
 	uint32_t priority;	/**< priority level 0 (low) to 10 (max) */
 	uint32_t period_mips;	/**< worst case instruction count per period */
 	uint32_t frames_per_sched;/**< output frames of pipeline, 0 is variable */
 	uint32_t xrun_limit_usecs; /**< report xruns greater than limit */
 
-	/* non zero if timer scheduled, otherwise DAI DMA irq scheduled */
-	uint32_t timer_delay;
+	/**< scheduling time domain */
+	enum sof_ipc_pipe_sched_time_domain time_domain;
```
Comment (plbossart) [inline include/sound/sof/topology.h]: I thought we agreed enums are not portable and need to be avoided in IPC structures?

Comment (tlauda) [reply include/sound/sof/topology.h]: @plbossart Fixed.

File: include/uapi/sound/sof/abi.h [hunk 2]
```diff
@@ -26,7 +26,7 @@
 
 /* SOF ABI version major, minor and patch numbers */
 #define SOF_ABI_MAJOR 3
-#define SOF_ABI_MINOR 3
+#define SOF_ABI_MINOR 4
```
Comment (plbossart) [inline include/uapi/sound/sof/abi.h]: is this really an ABI change if you just rename the fields?

Comment (tlauda) [reply include/uapi/sound/sof/abi.h]: Technically it's not, so it really depends what is our approach here. As Liam already stated in https://github.com/thesofproject/sof/pull/1131 there is no harm with increment. I can do it both ways, so your decision.

Comment (lgirdwood) [reply include/uapi/sound/sof/abi.h]: Probably best then not to bump ABI. I'll merge FW part once ABI removed.

Comment (tlauda) [reply include/uapi/sound/sof/abi.h]: @lgirdwood @plbossart Both PRs updated.

File: include/sound/sof/topology.h [hunk 3]
```diff
@@ -218,21 +218,25 @@ struct sof_ipc_comp_reply {
  * Pipeline
  */
```
Comment (dbaluta) [inline include/sound/sof/topology.h:220]: Hi @tlauda,

Sorry for being late. One minor comment here. Please use a more specific subject line. "Update" is just too generic.

For example, you can say ASoC: SOF: Align sof_ipc_pipe_new with FW code

Comment (tlauda) [reply include/sound/sof/topology.h:220]: @dbaluta Thanks for the suggestion. I'll try to do better next time.

**Pull request merged**